### PR TITLE
SecurityContext.init method

### PR DESCRIPTION
**Pull request by nenaraab**
With version 1.4.0, Pull Request #74 the [SecurityContext ](https://github.com/SAP/cloud-security-xsuaa-integration/blob/master/spring-xsuaa/src/main/java/com/sap/xs2/security/container/SecurityContext.java) was enhanced by a method to initialize the SecurityContextHolder with a JWT token as documented here:
https://github.com/SAP/cloud-security-xsuaa-integration/blob/master/spring-xsuaa/README.md#setup-security-context-for-non-http-requests

I'm not that happy with the released API (as of version 1.4.0) because the JWT token must not necessarily be a validated one. Only when people follows the documentation, then it is guaranteed. Please have a look at my code change proposal.

Overload SecurityContext.init method:
`SpringSecurityContext.init(String encodedJwtToken, JwtDecoder jwtDecoder, AuthoritiesExtractor authoritiesExtractor)`

From customer perspective this would be also a simplification.

### incompatible change:
- we face issues, when microservice uses SAP-libs, that use sap-internal container lib, that's why we need to move our functionality to a separate class, namely `SpringSecurityContext`

### CODE REVIEW

**Comment by nenaraab** [comment]
@strehle 
can you please perform an initial review?

**Comment by nenaraab** [comment]
@mwdb can you please perform a review?

File: spring-xsuaa/src/main/java/com/sap/xs2/security/container/SecurityContext.java [hunk 1]
```diff
@@ -32,36 +34,54 @@ static public UserInfo getUserInfo() throws UserInfoException {
...
+	 * @param extractLocalScopesOnly
+	 *            true when {@link Token#getAuthorities()} should only extract local
+	 *            scopes. Local scopes means that non-application specific scopes
+	 *            are filtered out and scopes are returned without appId prefix,
+	 *            e.g. "Display".
+	 * @deprecated use method
+	 *             {@link SpringSecurityContext#init(String, JwtDecoder, AuthoritiesExtractor)}
+	 *             instead
+	 */
+	@Deprecated
```
Comment (mwdb) [inline spring-xsuaa/src/main/java/com/sap/xs2/security/container/SecurityContext.java]: Why does this need to be deprecated?

**Comment by nenaraab** [comment]
Hi @mwdb 
Because it should not be used any longer :-) ! If people use it as documented, it’s fine. But if they just decode the token w/o validation, finally the security context will contain an invalid jwt...

**Pull request merged**